### PR TITLE
Automate HA install w/ Helm 3 and supplied RKE version

### PR DESCRIPTION
**Pull request by bmdepesa**
Creates load balancer, target groups, nodes, and creates/updates hostname in Route53.

### CODE REVIEW

File: tests/validation/tests/v3_api/test_create_ha.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    out = run_command_with_stderr(helm_init_cmd)
+    print(out)
+
+
+def install_rancher_self_signed():
+    helm_rancher_cmd = \
+        export_cmd + " && helm_v3 install rancher rancher-latest/rancher " + \
+        "--version " + RANCHER_CHART_VERSION + " " \
+        "--namespace cattle-system " + \
+        "--set hostname=" + RANCHER_HA_HOSTNAME
```
Comment (sangeethah) [inline tests/validation/tests/v3_api/test_create_ha.py]: @bmdepesa Can we have the rancher repo also parameterized so we are able to deploy from rancher-stable/rancher-alpha/rancher-latest ?

File: tests/validation/tests/v3_api/test_create_ha.py [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+from lib.aws import AmazonWebServices
+
+DATA_SUBDIR = os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                           'resource')
+
+RANCHER_CHART_VERSION = os.environ.get("RANCHER_CHART_VERSION")
+test_suffix = random_test_name("auto-ha")
+temp_hostname = test_suffix + ".qa.rancher.space"
+# if hostname is not provided, generate one ( to support onTag )
+RANCHER_HA_HOSTNAME = os.environ.get("RANCHER_HA_HOSTNAME", temp_hostname)
```
Comment (sangeethah) [inline tests/validation/tests/v3_api/test_create_ha.py]: @bmdepesa Can the test_suffix be an environment variable that user can set and be used for both nlb and route53 entry that gets created ? ".qa.rancher.space" can be got as another environment variable .

Comment (bmdepesa) [reply tests/validation/tests/v3_api/test_create_ha.py]: `qa.rancher.space` will never change as we use the hosted zone in Route53; I can parameterize the suffix and use that in place of the full host name (doing the concatenation in the test)

File: tests/validation/tests/v3_api/test_create_ha.py (hunk 1 above)
Comment (bmdepesa) [reply tests/validation/tests/v3_api/test_create_ha.py]: Added `RANCHER_HELM_REPO` -- user can supply one of `latest, stable, alpha`

File: tests/validation/tests/v3_api/test_create_ha.py (hunk 2 above)
Comment (bmdepesa) [reply tests/validation/tests/v3_api/test_create_ha.py]: Job now takes `RANCHER_HOSTNAME_PREFIX` and resources are now in format:
```
nlb-auto-21782-prefix
tg-80-auto-21782-prefix
tg-443-auto-21782-prefix
https://prefix.qa.rancher.space
```

**Comment by sangeethah** [comment]
LGTM

**Pull request merged**